src/interpreter/runtime/values/i32.js

```diff
@@ -1,14 +1,271 @@
+    return new i32(this._value | operand._value);
+  }
+
+  xor(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-ixor
+    return new i32(this._value ^ operand._value);
+  }
 
+  eqz(): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-ieqz
+    return new i32(this._value == 0 ? 1 : 0);
+  }
+
+  eq(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-ieq
+    return new i32(this._value == operand._value ? 1 : 0);
+  }
+
+  ne(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-ieq
+    return new i32(this._value != operand._value ? 1 : 0);
+  }
+
+  lt_u(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-lt-u
+    return new i32(
+      toUnsigned(this._value) < toUnsigned(operand._value) ? 1 : 0
+    );
+  }
+
+  lt_s(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-lt-s
+    return new i32(this._value < operand._value ? 1 : 0);
+  }
+
+  le_u(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-lt-u
+    return new i32(
+      toUnsigned(this._value) <= toUnsigned(operand._value) ? 1 : 0
+    );
+  }
+
+  le_s(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-lt-s
+    return new i32(this._value <= operand._value ? 1 : 0);
+  }
+
+  gt_u(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-gt-u
+    return new i32(
+      toUnsigned(this._value) > toUnsigned(operand._value) ? 1 : 0
+    );
+  }
+
+  gt_s(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-gt-s
+    return new i32(this._value > operand._value ? 1 : 0);
+  }
+
+  ge_u(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-gt-u
+    return new i32(
+      toUnsigned(this._value) >= toUnsigned(operand._value) ? 1 : 0
+    );
+  }
+
+  ge_s(operand: i32): i32 {
+    // https://webassembly.github.io/spec/core/exec/numerics.html#op-gt-s
+    return new i32(this._value >= operand._value ? 1 : 0);
+  }
+
+  equals(operand: i32): boolean {
+    return isNaN(this._value)
+      ? isNaN(operand._value)
+      : this._value == operand._value;
+  }
+
+  min(operand: i32): i32 {
+    return new i32(Math.min(this._value, operand._value));
+  }
+
+  max(operand: i32): i32 {
+    return new i32(Math.max(this._value, operand._value));
+  }
+
+  abs(): i32 {
+    return new i32(Math.abs(this._value));
+  }
+
+  neg(): i32 {
+    return new i32(-this._value);
+  }
+
+  copysign(operand: i32): i32 {
+    return new i32(
+      Math.sign(this._value) === Math.sign(operand._value)
+        ? this._value
+        : -this._value
+    );
+  }
+
+  toNumber(): number {
+    return this._value;
+  }
+
+  isTrue(): boolean {
+    return this._value == 1;
```

Comment:
We can use triple equal here. I'm not sure about this tbh, numeric types are not booleans, either the name is misleading or we could inline the `=== 1` in the callers?